deploy/device-operator.yaml

```diff
@@ -111,6 +111,17 @@ spec:
           description: VolStatus string that specifies the current state of the volume
             provisioning request.
           properties:
+            error:
+              description: Error denotes the error occurred during provisioning a
+                volume. Error field should only be set when State becomes Failed.
+              properties:
+                code:
+                  description: VolumeErrorCode represents the error code to represent
+                    specific class of errors.
+                  type: string
+                message:
+                  type: string
+              type: object
             state:
```

Comment:
Similar to above, API conventions mention about not using phase/state and there are KEPs for current resources to change this already. Each of the state should have it's own condition field  . https://github.com/kubernetes/kubernetes/issues/7856